docs/build/tutorials/wharfkit/howto_react.md

```diff
@@ -0,0 +1,231 @@
+---
+title: WharfKit - React Tutorial
+order: 101
+---
+
+# Using SessionKit
+
+To install the kit in a JavaScript or TypeScript (React) application:
+
+```bash
+yarn add @wharfkit/session @wharfkit/web-renderer @wharfkit/wallet-plugin-anchor
+```
+
+For a complete React example, check [this link](https://github.com/3dkrender/WAX_CS_Template).
+
+## Preparing the Session
+
+The code to prepare the session would look like this:
+
+```javascript
+// ... (previous code)
+
+const chains = [
+  {
+    id: import.meta.env.VITE_CHAINID,
+    url: import.meta.env.VITE_RPC,
+  }
+];
+
+const walletPlugins = [];
+walletPlugins.push(new WalletPluginAnchor());
+if (import.meta.env.VITE_CHAIN === 'mainnet') {
+  walletPlugins.push(new WalletPluginCloudWallet());
+}
+
+export const sessionKit = new SessionKit({
+  appName: import.meta.env.VITE_SITE_TITLE,
+  chains,
+  ui: new WebRenderer(),
+  walletPlugins,
+});
+
+// ... (rest of the code)
+```
+:::tip Note
+For the example, some data is taken from environment files, and their values will vary depending on the blockchain. For instance, to differentiate between mainnet and testnet:
+
+VITE_CHAIN=testnet  
+VITE_RPC=http://tapiwax.3dkrender.com  
+VITE_CHAINID=f16b1833c747c43682f4386fca9cbb327929334a762755ebec17f6f23c9b8a12
+
+VITE_CHAIN=mainnet  
+VITE_RPC=http://apiwax.3dkrender.com  
+VITE_CHAINID=1064487b3cd1a897ce03ae5b6a865651747e2e152090f99c1d19d44e01aea5a4
+:::
+
+[Sample Code](https://github.com/3dkrender/WAX_CS_Template/blob/main/Client/src/App.tsx)
+
+# Login
+
+To create our session instance, we need to call the login method from the session kit. We can use a button that, when pressed, triggers the call to the component. In the mentioned template, this component is called from the login button located in the main menu of the application 
+
+[Sample Code](https://github.com/3dkrender/WAX_CS_Template/blob/main/Client/src/components/Menu/Menu.tsx).
+
+The first step will be to retrieve the session kit instance that we have already created:
+
+```javascript
+import { sessionKit } from "../../App";
+```
+
+Every time a page of our application is reloaded, we can check if a user session already exists. If it does, we can retrieve it and avoid logging in again. This can be done using the useEffect hook:
+
+```javascript
+  const [session, setSession]: [Session | undefined, Dispatch<SetStateAction<Session | undefined>>] = useState<Session | undefined>(undefined);
+
+  useEffect(() => {
+    sessionKit.restore().then((session: Session | undefined) => {
```

Comment:
Might be good to pass the chain ID here into `restore()` so that as a developer, you're switching between mainnet and testnet, it knows which chain to restore. Otherwise you might end up restoring a mainnet account in a testnet environment! Not the end of the world... but figured I'd mention it 😉 

https://wharfkit.com/docs/sessionkit/restore#example-specific-chain

Reply:
Good tip!